src/Symfony/Component/VarExporter/LazyGhostTrait.php

```diff
@@ -29,16 +29,20 @@ trait LazyGhostTrait
      * properties and closures should accept 4 arguments: the instance to
      * initialize, the property to initialize, its write-scope, and its default
      * value. Each closure should return the value of the corresponding property.
+     * The special "\0" key can be used to define a closure that returns all
+     * properties at once when full-initialization is needed.
      *
      * Properties should be indexed by their array-cast name, see
      * https://php.net/manual/language.types.array#language.types.array.casting
      *
-     * @param \Closure(static):void|array<string, \Closure(static, string, ?string, mixed):mixed> $initializer
-     * @param array<string, true> $skippedProperties An array indexed by the properties to skip, aka the ones
-     *                                               that the initializer doesn't set when its a closure
+     * @param \Closure(static):void|array<string, \Closure(static, string, ?string, mixed):mixed>|array{"\0": \Closure(static):array} $initializer
```

Comment:
the structure of the array being returned by the default initializer should be documented

Reply:
This is described in the description. I don't see how we can describe this better. `array<string,mixed>` would look useless to me.

Reply:
`array<string, mixed>` is not useless. It would allow SA tools to warn you if you return a list instead of an array indexed by strings.

Reply:
OK, updated, and I figured out how to split the description on multiple lines.